bin/postgres_ha_admin

```diff
@@ -2,6 +2,41 @@
 
 require 'bundler/setup'
 require 'manageiq-postgres_ha_admin'
+require 'linux_admin'
+require 'logger'
+require '/var/www/miq/vmdb/config/environment'
+
+ManageIQ::PostgresHaAdmin.logger = Logger.new("/var/www/miq/vmdb/log/ha_admin.log")
+
+rails_handler = ManageIQ::PostgresHaAdmin::RailsConfigHandler.new(:file_path => "/var/www/miq/vmdb/config/database.yml", :environment => "production")
+rails_handler.before_failover do
+  LinuxAdmin::Service.new("evmserverd").stop
+end
+
+rails_handler.after_failover do |_new_conn_info|
+  LinuxAdmin::Service.new("evmserverd").restart
+  require "awesome_spawn"
+
+  rails_root = [
+    Pathname.new("/var/www/miq/vmdb"),
+    Pathname.new(File.expand_path(File.join(__dir__, "../..")))
+  ].detect { |f| File.exist?(f) }
+  AwesomeSpawn.run("rake evm:raise_server_event",
+                   :chdir  => rails_root,
+                   :params => ["--", {:event  => "db_failover_executed"}])
+end
 
 monitor = ManageIQ::PostgresHaAdmin::FailoverMonitor.new
+monitor.add_handler(rails_handler)
+
+conninfo = ActiveRecord::Base.connection.raw_connection.conninfo_hash.delete_blanks
+PglogicalSubscription.all.each do |s|
+  h = ManageIQ::PostgresHaAdmin::PglogicalConfigHandler.new(:subscription => s.id, :conn_info => conninfo)
+  h.after_failover do |new_conn_info|
+    PglogicalSubscription.find(h.subscription).delete
+    PglogicalSubscription.new(new_conn_info.slice(:dbname, :host, :user, :password, :port)).save
```

Comment:
This may be generic enough to do from the gem itself in the `PglogicalConfigHandler#write` method. As long as the subscription is removed in the `before_failover` that could be a possibility.

Reply:
Nope, it's not, we dynamically generate the subscription name in a specific way and will conditionally enable the extension as a part of saving the first subscription so it will be much easier to do this through the model.